document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -1186,6 +1186,89 @@ public void testFilterAndUnnest(String dataStoreName) throws IOException {
     assertDocsAndSizeEqual(dataStoreName, iterator, "query/unwind_filter_response.json", 7);
   }
 
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testUnnestWithRegularFilterAndNullAndEmptyPreservedAtSecondLevel(String dataStoreName)
+      throws IOException {
+    Collection collection = getCollection(dataStoreName);
+
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .addSelection(IdentifierExpression.of("item"))
+            .addSelection(IdentifierExpression.of("quantity"))
+            .addSelection(IdentifierExpression.of("sales.city"))
+            .addSelection(IdentifierExpression.of("sales.medium.type"))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("sales"), true))
+            .addFromClause(
+                UnnestExpression.builder()
+                    .identifierExpression(IdentifierExpression.of("sales.medium"))
+                    .preserveNullAndEmptyArrays(true)
+                    .filterTypeExpression(
```

Comment:
As discussed offline, `sales.medium.type` is not a valid access pattern for an array object, so it will result in two different resultsets between mongo / Postgres. So, to keep the result aligned in the test case, we have added at both places.